### ISSUE TITLE
Could java-coap using SSL with my own CA certificate?

### ISSUE THREAD

**Issue by KeithLaiKB** [PelionIoT/java-coap#57]
I am trying the condition that I have my own CA, and I try to use client to use SSL to connect the server with the CA certificate.
How could I implement it, I just could find the test about SSL using the secrete with keystore in this library.

**Comment by szysas** [comment]
Look at the tests, they are using own self sign CA. You would need to create own keystore.

**Comment by KeithLaiKB** [comment]
> Look at the tests, they are using own self sign CA. You would need to create own keystore.

I have a question is that, could the client just use the CA without keystore(.jks)?

**Comment by KeithLaiKB** [comment]
> > Look at the tests, they are using own self sign CA. You would need to create own keystore.
> 
> I have a question is that, could the client just use the CA without keystore(.jks)?My ca certificate format is .crt
> 
> and another question is that I want to use DTLS/SSL, but in the test (https://github.com/PelionIoT/java-coap/blob/26902e306995a439eb74335a3f35bcec74867d9c/coap-core/src/test/java/com/mbed/coap/transport/javassl/SSLUtils.java) it is about TLSv1.2, how can I do with DTLS/SSL?

**Comment by szysas** [comment]
No, you would need to make a keystore. There is very hand GUI tool to create keystores: https://keystore-explorer.org/

Java does not support DTLS so you would need to use some third party. 
In this fork: https://github.com/open-coap/java-coap, there is added integration for mbedtls that supports DTLS (PSK and Certificates): https://github.com/open-coap/java-coap/tree/master/mbedtls-transport

**Comment by KeithLaiKB** [comment]
so, you mean in master branch in this repository, it does not support DTLS directly and it does not support DTLS/SSL, right?

**Comment by KeithLaiKB** [comment]
> No, you would need to make a keystore. There is very hand GUI tool to create keystores: https://keystore-explorer.org/
> 
> Java does not support DTLS so you would need to use some third party. In this fork: https://github.com/open-coap/java-coap, there is added integration for mbedtls that supports DTLS (PSK and Certificates): https://github.com/open-coap/java-coap/tree/master/mbedtls-transport

so, you mean in master branch in this repository, it does not support DTLS directly and it does not support DTLS/SSL, right?

**Comment by szysas** [comment]
That's right

**Comment by szysas** [comment]
That's right, TLS is TCP based.
